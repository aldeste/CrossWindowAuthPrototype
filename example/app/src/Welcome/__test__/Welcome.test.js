import React from "react";
import Welcome from "../Welcome";
import renderer from "react-test-renderer";

jest.mock("../../Tags", () => ({
  Button: ({ className, children, ...props }) => (
    <button {...props}>{children}</button>
  ),
  Text: ({ className, children, ...props }) => <p {...props}>{children}</p>,
  Title: ({ className, children, ...props }) => <h1 {...props}>{children}</h1>,
  Form: ({ className, children, ...props }) => (
    <form {...props}>{children}</form>
  ),
  Label: ({ className, children, ...props }) => (
    <label {...props}>{children}</label>
  ),
  TextInput: ({ className, children, ...props }) => (
    <input {...props}>{children}</input>
  ),
  View: ({ className, children, ...props }) => <div {...props}>{children}</div>
}));

const mockClickFunction = jest.fn();

describe("Welcome react component", () => {
  const component = renderer.create(
    <Welcome
      title="This is title"
      username="This is username"
      onLogoutSubmit={e => mockClickFunction()}
    />
  );

  const tree = component.toJSON();

  it("Is a react component", () => {
    expect(typeof Welcome).toBe("function");
    expect(<Welcome />.$$typeof.toString()).toBe("Symbol(react.element)");
  });

  it("Matches earlier configuration", () => {
    expect(tree).toMatchSnapshot();
  });

  it("has a button", () => {
    expect(tree.children.find(el => el.type === "button")).toBeTruthy();
  });

  it("has a title to identify which welcome screen we're in", () => {
    expect(tree.children.find(el => el.type === "h1")).toBeTruthy();
  });

  it("has a text to identify which user is signed in", () => {
    expect(tree.children.find(el => el.type === "p")).toBeTruthy();
  });

  it("has an onClick function in the button", () => {
    tree.children.find(el => el.type === "button").props.onClick();
    expect(mockClickFunction).toHaveBeenCalled();
  });
});
