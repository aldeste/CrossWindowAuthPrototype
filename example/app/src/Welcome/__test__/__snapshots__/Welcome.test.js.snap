// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome react component Matches earlier configuration 1`] = `
<div>
  <h1>
    This is title
  </h1>
  <p>
    Here's some private information about you
  </p>
  <div>
    <h2>
      Yoda
    </h2>
    <p>
      Born: 
      896BBY
    </p>
    <p>
      Your eyes are 
      brown
    </p>
    <p>
      Your gender is 
      male
    </p>
    <h3>
      Your homeworld
    </h3>
    <div>
      <p>
        We don't know where you're from, just like some of the other characters. Maybe you know them?
      </p>
      <p>
        IG-88, Arvel Crynyd, Qui-Gon Jinn, R4-P17, Finn, Rey, Poe Dameron, BB8, Captain Phasma
      </p>
    </div>
  </div>
  <p>
    You can log out and try another user by clicking the log out button
  </p>
  <button
    alternative={true}
    onClick={[Function]}
  >
    Log out
  </button>
</div>
`;
